src/devices/Nmea0183/Identification.Extra.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+
+namespace Iot.Device.Nmea0183
+{
+    public partial class Identification : IEquatable<Identification>
+    {
+        /// <inheritdoc />
+        public bool Equals(Identification? other)
+        {
+            if (ReferenceEquals(null, other))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return shipNameField == other.shipNameField
+                   && callsignField == other.callsignField
+                   && mMSIField == other.mMSIField
+                   && calibrationDateField.Equals(other.calibrationDateField);
+        }
+
+        /// <inheritdoc />
+        public override bool Equals(object? obj)
+        {
+            if (ReferenceEquals(null, obj))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, obj))
+            {
+                return true;
+            }
+
+            if (obj.GetType() != GetType())
```

Comment:
any particular reason to block derived classes? I'd keep this to minimum: null check and refer to typed Equals

Reply:
Good catch, this code gets auto-generated by Resharper, and I never really thought of what this means.